syz-manager/repro_test.go

```diff
@@ -105,16 +97,76 @@ func TestReproOrder(t *testing.T) {
 	assert.Equal(t, crashes[2], obj.popCrash())
 }
 
+func TestReproRWRace(t *testing.T) {
+	mock := &reproMgrMock{
+		run: make(chan runCallback),
+	}
+	onlyOnce := true
+	obj := newReproManager(mock, 3, onlyOnce)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	go func() {
+		obj.Loop(ctx) // calls runRepro()
+	}()
+
+	assert.False(t, obj.CanReproMore())
+	obj.StartReproduction()
+	assert.True(t, obj.CanReproMore())
+
+	obj.Enqueue(&Crash{Report: &report.Report{Title: "A"}})
+	obj.Enqueue(&Crash{Report: &report.Report{Title: "A"}})
+	// Both crashes have the same title and onlyOnce is set, so only 1 is queued.
+
+	done := make(chan struct{})
+	go func() {
```

Comment:
I think you can omit ` go func() ` and the infinite loop here -- just `select` on `mock.run` to update `mock.reproProgExist`. `obj.Loop()` is already running in the background anyway.